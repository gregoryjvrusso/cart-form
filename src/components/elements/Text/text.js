import React from "react";
import { array, oneOfType, string } from "prop-types";
import { Container } from "./text.styles";

const Text = ({ children, ...props }) => (
  <Container data-testid="text" {...props}>{children}</Container>
);

Text.propTypes = {
  children: oneOfType([ array, string ]).isRequired,
};

Text.displayName = "Text";

export default Text;
