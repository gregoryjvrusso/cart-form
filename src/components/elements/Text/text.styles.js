import styled from 'styled-components';
import { typography, space, color } from 'styled-system';

export const Container = styled.p`
  ${typography}
  ${space}
  ${color}
`;
