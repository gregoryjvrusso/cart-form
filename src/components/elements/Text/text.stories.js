import React from "react";
import { storiesOf } from "@storybook/react";
import Text from "../Text";

storiesOf("Elements/Text", module)
  .add("default", () => (
    <div>
      <Text textAlign="left" color="green">
        Hello Left
      </Text>
      <Text textAlign="center" color="blue">
        Hello Center
      </Text>
      <Text textAlign="right">Hello Right</Text>
    </div>
  ))
  .add("Font Weight", () => (
    <div>
      <Text fontWeight="bold">Hello bold</Text>
      <Text fontWeight="normal">Hello normal</Text>
      <Text fontWeight="900">Hello 900</Text>
      <Text fontStyle="italic">Hello italic</Text>
    </div>
  ))
  .add("Color", () => (
    <div>
      <Text color="blue">Hello Blue</Text>
      <Text color="green">Hello Green</Text>
    </div>
  ));
