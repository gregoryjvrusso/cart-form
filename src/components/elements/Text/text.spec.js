import React from "react";
import { render } from "@testing-library/react";
import Text from "../Text";

describe("Text component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(<Text>Texto</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("verificar se os valores do css estÃ£o corretos", () => {
    const { getByTestId } = render(
      <Text color="#fff" fontSize="13px">
        Texto
      </Text>
    );
    expect(getByTestId("text")).toHaveStyle("color: #fff");
    expect(getByTestId("text")).toHaveStyle("font-size: 13px");
  });
});
