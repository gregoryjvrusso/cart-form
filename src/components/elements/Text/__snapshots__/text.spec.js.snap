// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component deve conseguir renderizar [snapshot] 1`] = `
<p
  class="sc-AxjAm dsBzxu"
  data-testid="text"
>
  Texto
</p>
`;
